ex2_hal/adcs/hardware_interface/source/adcs.c

```diff
@@ -19,6 +19,29 @@
 
 #include "adcs.h"
 
+ADCS_returnState HAL_ADCS_download_file_list_to_OBC(void){
+#ifdef ADCS_IS_STUBBED
+    return IS_STUBBED_A;
+#else
+    return ADCS_get_file_list();
+#endif
+}
+
+ADCS_returnState HAL_ADCS_download_file_to_OBC(adcs_file_download_id *id){
+#ifdef ADCS_IS_STUBBED
+    return IS_STUBBED_A;
+#else
+    // Spawn high-priority file download task
+    TaskHandle_t xHandle;
+    xTaskCreate(ADCS_download_file_task, "ADCS_download_file_task", ADCS_QUEUE_GET_TASK_SIZE, (void *)id, ADCS_QUEUE_GET_TASK_PRIO, &xHandle);
+
+    // Wait until task finishes and return
+    while(eTaskGetState(xHandle) != eDeleted);
```

Comment:
This should delay so it doesn't take up the CPU by constantly checking. Or add a semaphore that the task gives right before it deletes itself. The semaphore can then be free also

Reply:
The semaphore can be passed using your id struct

Reply:
I'm going to make it return immediately actually since the operation takes too long and the ground station times out. Will save the result using sys_log